src/lib/Scenes/MyCollection/Screens/Artwork/Components/MyCollectionArtworkHeader.tsx

```diff
@@ -24,6 +24,55 @@ const MyCollectionArtworkHeader: React.FC<MyCollectionArtworkHeaderProps> = (pro
 
   const isImage = (toCheck: any): toCheck is Image => !!toCheck
 
+  const imageIsProcessing = (image: Image | null) => {
+    if (!image) {
+      return false
+    }
+
+    const isProcessing = image.height === null
+    return isProcessing
+  }
+
+  const hasImagesStillProcessing = (mainImage: any, imagesToCheck: MyCollectionArtworkHeader_artwork["images"]) => {
+    if (!isImage(mainImage) || imageIsProcessing(mainImage)) {
+      return true
+    }
+
+    if (!imagesToCheck) {
+      return false
+    }
+
+    const concreteImages = imagesToCheck as Image[]
+    const stillProcessing = concreteImages.filter((image) => imageIsProcessing(image)).length > 0
+    return stillProcessing
+  }
+
+  const processingText = () => {
+    return images && images?.length > 0 ? "Processing photos" : "Processing photo"
+  }
+
+  const MainImageView = () => {
```

Comment:
We shouldn't be creating new react component types during render, since that makes react confused about what has and hasn't changed
```suggestion
  const renderMainImageView = () => {
```

and then below

```diff
- <MainImageView />
+ {renderMainImageView()}
```